File: DrawingLayer/ViewController.m
//
//  ViewController.m
//  DrawingLayer
//
//  Created by wangwenjian on 2022/3/29.
//

#import "ViewController.h"
#import "TrapezoidLayer.h"
@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    TrapezoidLayer *layer = [TrapezoidLayer layer];
    [self.view.layer addSublayer:layer];
}


@end

File: Present_Demo/ViewController.swift
//
//  ViewController.swift
//  Present_Demo
//
//  Created by wangwenjian on 2022/5/18.
//

import UIKit

class ViewController: UIViewController {

    
    let yellowVC = UIViewController()
    let blueVC = UIViewController()

    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        
        self.view.backgroundColor = .red
        
        DispatchQueue.main.asyncAfter(deadline: .now() + 5) {
            self.yellowVC.view.backgroundColor = .yellow
            self.present(self.yellowVC, animated: true) {
                
            }
            DispatchQueue.main.asyncAfter(deadline: .now() + 15) {
                let vc = self.yellowVC.presentedViewController
                vc?.dismiss(animated: true, completion: {
                    self.yellowVC.dismiss(animated: true) {
        
                    }
                })

            }
        }

        DispatchQueue.main.asyncAfter(deadline: .now() + 10) {
            self.blueVC.view.backgroundColor = .blue
            self.yellowVC.present(self.blueVC, animated: true) {
                
            }
        }
    }


}

